snuba/datasets/events.py

```diff
@@ -392,4 +393,4 @@ def get_extensions(self) -> Mapping[str, QueryExtension]:
         }
 
     def get_query_processors(self) -> Sequence[QueryProcessor]:
-        return [BasicFunctionsProcessor(), PrewhereProcessor()]
+        return [BasicFunctionsProcessor(), ApdexProcessor(), PrewhereProcessor()]
```

Comment:
We don't currently have any use cases for generating apdex on non-transaction events. The only use case we have currently is on the `duration` column in the transactions table. In sentry we can enforce that is the only column it is used on.

Reply:
I like that this restricts these functions to the 2 datasets. Should we raise an exception if you try to pass it to another dataset without running the query.

Reply:
About events, ok, will remove from events
About the error. Clickhouse will fail saying that it does not know what apdex is, so it is not a very different error message than the one we would send. On the other hand in order for us to do the check we would have als oto keep a blacklist somewhere to distinguish the functions we do not want to send to clickhouse from those we want to.